drivers/ethernet/eth_nxp_enet.c

```diff
@@ -163,12 +163,15 @@ static inline void ts_register_tx_event(const struct device *dev,
 	struct net_pkt *pkt = frameinfo->context;
 
 	if (pkt && atomic_get(&pkt->atomic_ref) > 0) {
-		if (eth_get_ptp_data(net_pkt_iface(pkt), pkt) && frameinfo->isTsAvail) {
+		if ((eth_get_ptp_data(net_pkt_iface(pkt), pkt) ||
+		     net_pkt_is_tx_timestamping(pkt)) &&
+		    frameinfo->isTsAvail) {
```

Comment:
How about keep only `if (frameinfo->isTsAvail)` ? Because during TX, it had already do the same filtering.